### DISCUSSION TITLE
Zitald and the Entra ID intregration - Cant get family_name or given_name autofilled in register form

### DISCUSSION THREAD

**Discussion by noerdar** [zitadel/zitadel#8935] [Q&A]
Hi Community

I have setup my Zitadel to use Entra ID, the template, not the SAML. It works, but even I have setup claims on the App registration, it will not prefill them. I tried with this scrip as an action for Post Authentication, but it does not fill out the given name or family name. This is not an Okta user, but a Azure Entra user, but I think it would be the same.

/**
 * Set first and lastname of a user on just in time provisioning for okta.
 * Useful if you like to fill the first and lastname with the name stored on okta, so the user doesn't have to fill himself.
 * Also set email to verified, so the user doesn't get a verification email
 *
 * Flow: External Authentication, Trigger: Post Authentication
 *
 * @param ctx
 * @param api
 */
let logger = require("zitadel/log")

function mapOktaOauth(ctx, api) {
  logger.log('Populating extra information for new Okta user');

  console.log('running');

  if (ctx.v1.externalUser.externalIdpId != "292730930177507332") {
    return
  }
  api.setFirstName(ctx.v1.providerInfo.given_name);
  api.setLastName(ctx.v1.providerInfo.family_name);
  api.setEmailVerified(true)
  api.setEmail(ctx.v1.providerInfo.email)
  api.setPreferredUsername(ctx.v1.providerInfo.preferred_username)
}

**Comment by hifabienne** [comment]
Hi @noerdar 
Are you self hosting or in the cloud?
If self hosting do you see the log? So that you can confirm the action actually runs?

**Reply by hifabienne** [reply]
Generally if you use the microsoft template for entra id, you should get all the data automatically, no need to add an action for this

**Reply by hifabienne** [reply]
only thing you would need is the isemailverified, as this information is not provided by entra id
